core/src/test/java/org/elasticsearch/common/network/NetworkModuleTests.java

```diff
@@ -99,64 +130,68 @@ public void testRegisterTransport() {
         Settings settings = Settings.builder().put(NetworkModule.TRANSPORT_TYPE_KEY, "custom")
             .put(NetworkModule.HTTP_ENABLED.getKey(), false)
             .build();
-        NetworkModule module = new NetworkModule(new NetworkService(settings, Collections.emptyList()), settings, false);
-        module.registerTransport("custom", FakeTransport.class);
-        assertBinding(module, Transport.class, FakeTransport.class);
+        NetworkModule module = new NetworkModule(settings, false);
+        FakeTransportFactory custom = new FakeTransportFactory("custom");
+        module.registerTransport(custom);
```

Comment:
The nice thing about making `NetworkModule` take a list of plugins is that these register method can become implementation details/non-public/go away entirely. You'd need to make a anonymous inner class here for the plugin but it isn't a big deal I think.